File: frontend/src/Components/Popups/CreateYoutubePlaylistPopup.js
import React, { useState } from "react";
import server from '../Static/Constants';

const CreateYoutubePlaylistPopup = ({ onClose }) => {
    const [playlistTitle, setPlaylistTitle] = useState("");
    const [playlistDescription, setPlaylistDescription] = useState("");
    const [errorMessage, setErrorMessage] = useState(null);
    const [successMessage, setSuccessMessage] = useState(null);

    const handleCreatePlaylist = async () => {
        setErrorMessage(null);
        setSuccessMessage(null);

        if (!playlistTitle || !playlistDescription) {
            setErrorMessage("Please fill in all fields.");
            return;
        }

        try {
            const token = localStorage.getItem("token");
            const response = await fetch(`${server}/create/youtube_playlist/`, {
                method: "POST",
                headers: {
                    "Authorization": token,
                    "Content-Type": "application/json"
                },
                body: JSON.stringify({
                    title: playlistTitle,
                    description: playlistDescription,
                })
            });

            if (response.ok) {
                setSuccessMessage("Playlist created successfully!");
                setPlaylistTitle("");
                setPlaylistDescription("");
            } else {
                const errorData = await response.json();
                setErrorMessage(errorData.message || "An error occurred during playlist creation.");
            }
        } catch (error) {
            console.error("Error creating playlist:", error);
            setErrorMessage("An error occurred during playlist creation.");
        }
    };

    return (
        <div className="popup-overlay">
            <div className="popup-content">
                <h2>Create YouTube Playlist</h2>
                
                {errorMessage && <p className="error-message" style={{ color: "red" }}>{errorMessage}</p>}
                {successMessage && <p className="success-message" style={{ color: "green" }}>{successMessage}</p>}
                
                <input
                    type="text"
                    placeholder="Playlist Title"
                    value={playlistTitle}
                    onChange={(e) => setPlaylistTitle(e.target.value)}
                />
                <textarea
                    placeholder="Playlist Description"
                    value={playlistDescription}
                    onChange={(e) => setPlaylistDescription(e.target.value)}
                />
                <button onClick={handleCreatePlaylist}>Create Playlist</button>
                <button onClick={onClose}>Cancel</button>
            </div>
        </div>
    );
};

export default CreateYoutubePlaylistPopup;

File: flutter_frontend/lib/assets/youtube/popups/createYoutubePlaylistPopup.dart
import 'dart:convert';
import 'package:flutter/material.dart';
import 'package:flutter_secure_storage/flutter_secure_storage.dart';
import 'package:http/http.dart' as http;
import 'package:flutter_frontend/static/constants.dart';

class CreateYoutubePlaylistPopup extends StatefulWidget {
  final VoidCallback onClose;

  const CreateYoutubePlaylistPopup({Key? key, required this.onClose})
    : super(key: key);

  @override
  State<CreateYoutubePlaylistPopup> createState() =>
      _CreateYoutubePlaylistPopupState();
}

class _CreateYoutubePlaylistPopupState
    extends State<CreateYoutubePlaylistPopup> {
  final _storage = const FlutterSecureStorage();

  final TextEditingController _titleController = TextEditingController();
  final TextEditingController _descriptionController = TextEditingController();

  String? errorMessage;
  String? successMessage;
  bool loading = false;

  Future<void> handleSubmit() async {
    final title = _titleController.text.trim();
    final description = _descriptionController.text.trim();

    if (title.isEmpty || description.isEmpty) {
      setState(() {
        errorMessage = "Please fill in all fields.";
        successMessage = null;
      });
      return;
    }

    setState(() {
      loading = true;
      errorMessage = null;
      successMessage = null;
    });

    try {
      final token = await _storage.read(key: 'token');
      if (token == null) throw Exception("Not authenticated");

      final res = await http.post(
        Uri.parse('$server/create/youtube_playlist/'),
        headers: {"Authorization": token, "Content-Type": "application/json"},
        body: jsonEncode({"title": title, "description": description}),
      );

      if (res.statusCode == 200) {
        setState(() {
          successMessage = "Playlist created successfully!";
          errorMessage = null;
        });

        _titleController.clear();
        _descriptionController.clear();
      } else {
        final data = jsonDecode(res.body);
        setState(() {
          errorMessage =
              data["message"] ?? "An error occurred during playlist creation.";
        });
      }
    } catch (e) {
      setState(() {
        errorMessage = "An error occurred during playlist creation.";
      });
    } finally {
      setState(() => loading = false);
    }
  }

  @override
  Widget build(BuildContext context) {
    return AlertDialog(
      title: const Text("Create YouTube Playlist"),
      content: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          TextField(
            controller: _titleController,
            decoration: const InputDecoration(labelText: "Playlist Title"),
          ),
          const SizedBox(height: 12),
          TextField(
            controller: _descriptionController,
            maxLines: 3,
            decoration: const InputDecoration(
              labelText: "Playlist Description",
            ),
          ),
          if (errorMessage != null)
            Padding(
              padding: const EdgeInsets.only(top: 12),
              child: Text(
                errorMessage!,
                style: const TextStyle(color: Colors.red),
              ),
            ),
          if (successMessage != null)
            Padding(
              padding: const EdgeInsets.only(top: 12),
              child: Text(
                successMessage!,
                style: const TextStyle(color: Colors.green),
              ),
            ),
        ],
      ),
      actions: [
        TextButton(onPressed: widget.onClose, child: const Text("Cancel")),
        ElevatedButton(
          onPressed: loading ? null : handleSubmit,
          child: loading
              ? const SizedBox(
                  width: 16,
                  height: 16,
                  child: CircularProgressIndicator(strokeWidth: 2),
                )
              : const Text("Create Playlist"),
        ),
      ],
    );
  }
}
